### ISSUE TITLE
Health check endpoint

### ISSUE THREAD

**Issue by astrauka** [andypetrella/spark-notebook#751]
Notebook servers sometimes starts to malfunction - the web interface is up and running, but kernels are not running and cannot be created or restarted.
Is there an endpoint to check the health stats such that we know when to restart the server.
Just hitting the Web UI will not work as it is responsive.
@vidma @andypetrella

**Comment by vidma** [comment]
that's quite valid use-case.  so you'd automatically reload the notebook in such a case?

however, a thing to have in mind is that some malfunctionings are fatal:
- e.g. we've seen `out of memory [RAM]` on driver/spark-notebook machine, if too many notebooks are open
- do you remember other use-cases?

**Comment by astrauka** [comment]
@vidma 
> that's quite valid. so you'd automatically reload the notebook in such a case?

Yes, we would trigger a reload by restarting a process. The reload procedure would not be a part of notebooks functionality as that's a monitoring job. But monitoring needs to know when to reload.

> we've seen out of memory [RAM] on driver/spark-notebook machine, if too many notebooks are open

If the notebooks process crashes, then the newly created health check endpoint will not respond and monitoring will restart notebooks

What we need is a health check endpoint that could tell if server is in a good state, which means it can host Web UI and start new kernels
